sensor/common/scannerclient/grpc_client.go

```diff
@@ -2,72 +2,172 @@ package scannerclient
 
 import (
 	"context"
+	"fmt"
 	"strings"
 
 	"github.com/pkg/errors"
+	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
 	"github.com/stackrox/rox/generated/storage"
 	"github.com/stackrox/rox/pkg/clientconn"
+	"github.com/stackrox/rox/pkg/env"
 	"github.com/stackrox/rox/pkg/images/utils"
 	"github.com/stackrox/rox/pkg/logging"
 	"github.com/stackrox/rox/pkg/mtls"
 	"github.com/stackrox/rox/pkg/registries/types"
 	scannerV1 "github.com/stackrox/scanner/generated/scanner/api/v1"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/status"
 )
 
 var (
 	log = logging.LoggerForModule()
 )
 
-// Client is a Scanner gRPC Client.
-type Client struct {
-	client scannerV1.ImageScanServiceClient
-	conn   *grpc.ClientConn
+// ScannerClient is the interface to remote image scanners used by Sensor.
+type ScannerClient interface {
+	GetImageAnalysis(context.Context, *storage.Image, *types.Config) (*ImageAnalysis, error)
+	Close() error
 }
 
-// dial Scanner and return a new Client.
-// dial is non-blocking and returns a non-nil error upon configuration error.
-func dial(endpoint string) (*Client, error) {
-	if endpoint == "" {
-		return nil, errors.New("Invalid Scanner endpoint (empty)")
+// ImageAnalysis is the result of an image analysis.
+type ImageAnalysis struct {
```

Comment:
I wonder if it would be cleaner to make this an interface which is implemented by some Scanner v2 client and Scanner v4 client instead of putting the data of both of them into one struct and then determining which data to use based on which field is not nil